### PR TITLE
feat: Add platform APIs to My report page [3]

### PR DESCRIPTION
**Pull request by bistaastha**
UI Preview:
Screenshot 2024-05-13 at 8 09 05 AM

### CODE REVIEW

File: src/app/core/models/platform/v1/reports-query-params.model.ts [hunk 1]
```diff
@@ -5,4 +5,6 @@ export interface ReportsQueryParams {
   order?: string;
   id?: string;
   next_approver_user_ids?: string;
+  q?: string;
```
Comment (bistaastha) [inline src/app/core/models/platform/v1/reports-query-params.model.ts:8]: Typing fix in follow up

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/fyle/my-reports/my-reports.page.ts [hunk 2]
```diff
@@ -467,11 +476,11 @@ export class MyReportsPage {
     this.filterPills = this.generateFilterPills(this.filters);
   }
 
-  onReportClick(erpt: ExtendedReport): void {
-    this.router.navigate(['/', 'enterprise', 'my_view_report', { id: erpt.rp_id, navigateBack: true }]);
+  onReportClick(erpt: Report): void {
```
Comment (mvaishnavi) [inline src/app/fyle/my-reports/my-reports.page.ts:479]: rename erpt to report across this file.

File: src/app/shared/components/reports-card/reports-card.component.html [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <div *ngIf="showDate" class="reports-card--date">
-  {{ erpt.rp_created_at | date : 'MMM dd, YYYY' }}
+  {{ erpt.created_at | date : 'MMM dd, YYYY' }}
```
Comment (mvaishnavi) [inline src/app/shared/components/reports-card/reports-card.component.html:2]: `report.`

File: src/app/shared/components/reports-card/reports-card.component.ts [hunk 4]
```diff
@@ -1,24 +1,27 @@
...
+import { Report } from 'src/app/core/models/platform/v1/report.model';
 
 @Component({
   selector: 'app-reports-card',
   templateUrl: './reports-card.component.html',
   styleUrls: ['./reports-card.component.scss'],
 })
 export class ReportsCardComponent implements OnInit {
-  @Input() erpt: ExtendedReport;
+  @Input() erpt: Report;
```
Comment (mvaishnavi) [inline src/app/shared/components/reports-card/reports-card.component.ts:11]: `report:`

**Review by mvaishnavi — CHANGES_REQUESTED** [review]
fix the erpt to report across.

**Review by mvaishnavi — APPROVED** [review]

**Pull request closed without merging**